subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/TestListenerBuildOperationAdapter.java

```diff
@@ -0,0 +1,150 @@
+    private final BuildOperationExecutor buildOperationExecutor;
+
+    public TestListenerBuildOperationAdapter(BuildOperationExecutor buildOperationExecutor, BuildOperationListener listener, BuildOperationIdFactory buildOperationIdFactory, Clock clock) {
+        this.buildOperationExecutor = buildOperationExecutor;
+        this.listener = listener;
+        this.buildOperationIdFactory = buildOperationIdFactory;
+        this.clock = clock;
+    }
+
+    @Override
+    public void started(final TestDescriptorInternal testDescriptor, TestStartEvent startEvent) {
+        BuildOperationDescriptor testBuildOperationDescriptor = createTestBuildOperationDescriptor(testDescriptor, startEvent);
+        long currentTime = clock.getCurrentTime();
+        runningTests.put(testDescriptor, new InProgressExecuteTestBuildOperation(testBuildOperationDescriptor, currentTime));
+        listener.started(testBuildOperationDescriptor, new OperationStartEvent(currentTime));
+    }
+
+    @Override
+    public void completed(TestDescriptorInternal testDescriptor, TestResult testResult, TestCompleteEvent completeEvent) {
+        InProgressExecuteTestBuildOperation runningOp = runningTests.remove(testDescriptor);
+        listener.finished(runningOp.descriptor, new OperationFinishEvent(runningOp.startTime, clock.getCurrentTime(), testResult.getException(), new BuildOperationTestResult(testResult)));
+    }
+
+    @Override
+    public void output(final TestDescriptorInternal testDescriptor, final TestOutputEvent event) {
+        InProgressExecuteTestBuildOperation runningOp = runningTests.get(testDescriptor);
+        BuildOperationDescriptor.Builder outputDescription = BuildOperationDescriptor.displayName(event.getDestination().name())
+            .operationType(BuildOperationCategory.TASK)
+            .details(TEST_OUTPUT_DETAILS);
+        BuildOperationDescriptor outputBuildOperationDescriptor = outputDescription.build(newOperationIdentifier(), runningOp.descriptor.getId());
+        long currentTime = clock.getCurrentTime();
+        listener.started(outputBuildOperationDescriptor, new OperationStartEvent(currentTime));
+        listener.finished(outputBuildOperationDescriptor, new OperationFinishEvent(currentTime, currentTime, null, new DefaultTestOutputBuildOperationResult(event)));
+    }
+
+    private BuildOperationDescriptor createTestBuildOperationDescriptor(TestDescriptor testDescriptor, TestStartEvent testStartEvent) {
+        DefaultTestBuildOperationDetails details = new DefaultTestBuildOperationDetails(testDescriptor, testStartEvent.getStartTime());
+        InProgressExecuteTestBuildOperation parentOperation = runningTests.get(testDescriptor.getParent());
+        Object parentId = parentOperation == null ? buildOperationExecutor.getCurrentOperation().getId() : parentOperation.descriptor.getId();
+        return BuildOperationDescriptor.displayName(testDescriptor.getName())
+            .operationType(BuildOperationCategory.TASK)
```

Comment:
❌ This is not a task, so it seems this is the wrong category.

Reply:
I thought task category would fit here as its running in the context of a task. removed the category